Add modal window for email signup

alexsanford opened the pull request:
This is based on "Version 4" of the designs here: p4CBkD-JP-p2

Before merging, at the very least we need to do the following:

- [x] Only show the modal on initial activation.
- [x] Update the code to push to our production mailing list, not the test list.

## Testing instructions

- Switch to this branch and load any page in Sensei.
- The modal window should pop up.
- Before testing various cases, reload the page and ensure the modal does not pop up the second time, regardless of whether it was submitted. Also, deactivate and reactivate Sensei, and ensure the modal does not appear.
- To see the modal again, deactivate Sensei, delete the option `sensei_installed` from the `wp_options` table, and reactivate Sensei.
- In the modal signup, try entering a valid email, invalid email, no email, the same email that was already submitted, etc.
- Note that this is submitting email addresses to the production email list, so feel free to delete them (or ask someone to do so) after you are done testing.

## Things to discuss (Edit: these are discussed below)

- The design should be close to the P2 post, but feel free to suggest tweaks.
- I'm using a MailChimp form that behaves in the default way, which is to open the MailChimp page in a new tab. This seemed to be the simplest way to do it.
  - Are we ok with this functionality?
  - If so, we should confirm that the information on that page is correct, and the links/buttons go where we want them to.
- I feel that it would be better for this form to be hosted outside of the plugin somehow (perhaps on SenseiLMS.com) and then displayed either using JS or an iframe. This will add some development time, but may be a good long-term decision. That said, the benefit may not be huge (this form will only be shown on initial activation, so having the flexibility to change it quickly may not be very helpful).

lkraav commented:
Is this meant to be unconditionally always loaded? I don't see any `if` conditions.

alexsanford commented:
@lkraav No, it's still a work-in-progress. I'm just saving it here while I work on it. Still more to be done before we ship it, including making it only show up on initial activation of the plugin 🙂

jom commented:
Is there any way we can transmit the email address to MC in the background (either directly from plugin or via senseilms.com)? I feel like the popup submit will be a blocker for a lot of people. I have default settings on Chrome and it blocked the submit.

donnapep commented:
This is looking great, and good call on the GDPR text!

Some minor things:
- Could the placeholder text in the text field be changed to non-uppercase (i.e. Enter your email address)?
- The email address should default to the email that's in the _General_ WordPress settings.
- Could we add a reCAPTCHA to the signup form? There's an option in Mailchimp somewhere.
- We should style the Mailchimp pages a bit. Could we try a blue background (#00ace6) for the entire page with white text similar to what WooCommerce does?

![screen shot 2019-02-20 at 6 06 16 am](https://user-images.githubusercontent.com/1190420/53087822-0ef39180-34d6-11e9-8515-d2e2ef5e2d99.jpg)

> This will add some development time, but may be a good long-term decision.
Given that this pop-up will likely be moved into an onboarding wizard in the future, our time horizon is more short-term. Probably best to think of the modal as a transient thing that will go away at some point.

alexsanford commented:
> I feel like the popup submit will be a blocker for a lot of people. I have default settings on Chrome and it blocked the submit.

@jom Interesting...this is the default behaviour for MailChimp, and WooCommerce's onboarding has the exact same behaviour. Does your Chrome block the window for WooCommerce as well?

alexsanford commented:
@donnapep 

> Could the placeholder text in the text field be changed to non-uppercase (i.e. Enter your email address)?

Done 6deec70e

> The email address should default to the email that's in the _General_ WordPress settings.

Done 3a2a65b2

> Could we add a reCAPTCHA to the signup form? There's an option in Mailchimp somewhere.

Done

> We should style the Mailchimp pages a bit. Could we try a blue background (#00ace6) for the entire page with white text and buttons similar to what WooCommerce does?

I took a stab at this, but I don't seem to have as much control as Woo does (maybe because we're on a free account?). I changed the background and such, but I couldn't change the top heading or the GDPR text (go to [this link](https://senseilms.us19.list-manage.com/subscribe/post?u=7a061a9141b0911d6d9bafe3a&id=278a16a5ed) to see the full form - users will be directed here after the Captcha if there is an error).

> A question on flexibility though...what sorts of changes would we not be able to make quickly? I'm assuming we'd still be able to log in to Mailchimp to tweak text and colours?

From within Mailchimp we can change everything on the landing page after the modal is submitted. The modal itself is currently entirely within the plugin, so we would not be able to change text/colors/fields without releasing a new version (unless we host it elsewhere).

jom commented:
@alexsanford 

> Interesting...this is the default behaviour for MailChimp, and WooCommerce's onboarding has the exact same behaviour. Does your Chrome block the window for WooCommerce as well?

I tried this with several Jurassic Ninjas. First time I tried it, WooCommerce's popup was also blocked. I turned off and back on the pop-up block setting in Chrome and it started working. Strange! I wonder if it is a bug in Chrome. I've also been having some blocks with CMD-clicking, so I wonder if something is off with Chrome.

donnapep commented:
> I took a stab at this, but I don't seem to have as much control as Woo does (maybe because we're on a free account?). I changed the background and such, but I couldn't change the top heading or the GDPR text (go to [this link](https://senseilms.us19.list-manage.com/subscribe/post?u=7a061a9141b0911d6d9bafe3a&id=278a16a5ed) to see the full form - users will be directed here after the Captcha if there is an error).

OK, let me look and see if I can figure out what Woo is doing in MC.

> The modal itself is currently entirely within the plugin, so we would not be able to change text/colors/fields without releasing a new version (unless we host it elsewhere).

I'm OK with that if you are. 🙂

Is this still a WIP? Let me know when it's ready for a full review and I'll revisit. Thx!

donnapep commented:
@alexsanford I'm wondering if we'll even be able to create a new signup form for the existing Sensei list. I'm not sure it's possible? Do you know how this will work?

alexsanford commented:
> @alexsanford I'm wondering if we'll even be able to create a new signup form for the existing Sensei list. I'm not sure it's possible? Do you know how this will work?

Noting here for the record that we discussed this on Slack. For now, we will use the same signup form for both the plugin and the SenseiLMS.com signup. We may want to move one to a MailChimp Landing Page in the future.

alexsanford commented:
@donnapep @jom I've removed the WIP, this is ready for testing 🙂 

Note that I've added testing instructions to the description above.

Changes from before:

- Emails are submitted to the production MailChimp list.
- Pop-up only happens once.
- Strings in the template are internationalized.

donnapep commented:
I should have specified on the card that this pop-up should only be triggered for new users on activation if they haven't already seen it. Existing users would have automatically been added to the WooCommerce mailing list when they purchased Sensei, so no need to bother them again.

alexsanford commented:
@donnapep Thanks for the clarification! This is fixed in https://github.com/Automattic/sensei/pull/2450/commits/547fd3030a3341d2d4764f7feeb66073d091c53d, and I've updated the testing instructions 🙂

donnapep commented on includes/email-signup/class-sensei-email-signup-form.php:
```diff
@@ -0,0 +1,112 @@
+		wp_register_script(
+			'jquery-modal',
+			'https://cdnjs.cloudflare.com/ajax/libs/jquery-modal/0.9.1/jquery.modal.min.js',
+			false,
```
Seems that jQuery modal would be dependent on jQuery.

alexsanford replied:
Good catch! https://github.com/Automattic/sensei/pull/2450/commits/ed6ca6b2087f7dcca9194936c918f7b40dc9cd95

alexsanford commented on assets/css/admin/email-signup.scss:
```diff
@@ -0,0 +1,56 @@
+#mc_embed_signup {
+	display: none;
+	border-radius: 0;
+	font-family: Futura;
```
Done in https://github.com/Automattic/sensei/pull/2450/commits/ea5ac4801a575df2b9dc73bcb4a468c8e235c4f0

alexsanford commented on assets/js/admin/email-signup.js:
```diff
@@ -0,0 +1,24 @@
+			fadeDuration: 250,
+			showClose: false,
+		} );
+	}, 1000 );
```
I tried removing the timeout, and it felt too quick to me. I've lowered it in https://github.com/Automattic/sensei/pull/2450/commits/59ec04a961949109cfa0cea460e636293aa8611c to 250 which is quicker, but doesn't feel rushed. What do you think?

donnapep replied:
Yeah, 250 seems reasonable. 👍

alexsanford commented:
Note that I did a rebase and force-push to fix conflicts. This means that the plugin file is changed from `woothemes-sensei.php` to `sensei.php` and the plugin will have to be reactivated.

donnapep commented:
Sorry, one more thing. Could we bump the text to at least 16px? The header could go up proportionately as well. It's a bit hard to read right now. Once that's adjusted this can
